File: iosApp/iosApp/Theme/Color.swift
import SwiftUI

// MARK: - Silent Waters Color Palette
// Dewy blues and weathered gray merge into a hushed palette,
// like drifting fog on silent waters.

extension Color {
    // MARK: - Base Palette Colors

    static let lightCyan = Color(hex: "E0FBFC")      // Delicate, sunlit blue tint
    static let lightBlue = Color(hex: "C2DFE3")       // Soft pastel blue
    static let coolSteel = Color(hex: "9DB4C0")       // Cool, steely blue with hint of mist
    static let blueSlate = Color(hex: "5C6B73")       // Blue with hints of cool authority
    static let jetBlack = Color(hex: "253237")        // Intense darkness with enigmatic appeal

    // MARK: - Success/Confirmation Colors

    static let seaGlass = Color(hex: "8BB5A1")        // Muted teal-green for success states
    static let sageGreen = Color(hex: "6A8E7F")       // Deeper confirmation color

    // MARK: - Warning/Attention Colors

    static let driftwood = Color(hex: "C9B896")       // Warm sand tone for warnings
    static let weatheredBrass = Color(hex: "A89968")  // Deeper attention color

    // MARK: - Map/Route Colors

    static let coastalPath = Color(hex: "7A9CAF")     // Active route/current trip
    static let harborBlue = Color(hex: "5C8AA8")      // Historical routes
    static let mistyLavender = Color(hex: "A8B5C7")   // Alternative path suggestion

    // MARK: - Category/Tag Colors

    static let stoneGray = Color(hex: "8C979E")       // Neutral category
    static let seafoamTint = Color(hex: "A8D5D8")     // Water-related trips
    static let duskPurple = Color(hex: "8A90A6")      // Evening/night activities
    static let sunrisePeach = Color(hex: "D4B5A8")    // Morning activities

    // MARK: - Disabled/Inactive States

    static let mist = Color(hex: "D8E2E6")            // Disabled text/icons (light mode)
    static let charcoal = Color(hex: "3A4449")        // Disabled text/icons (dark mode)

    // MARK: - Data Visualization Gradient

    static let gradientStart = lightCyan              // (E0FBFC)
    static let gradientStep1 = lightBlue              // (C2DFE3)
    static let gradientStep2 = coolSteel              // (9DB4C0)
    static let gradientStep3 = coastalPath            // (7A9CAF)
    static let gradientEnd = blueSlate                // (5C6B73)

    // MARK: - Semantic Colors

    // Primary colors (Cool Steel for tranquility and trust)
    static let primaryLight = coolSteel
    static let primaryDark = lightBlue

    // Container colors
    static let primaryContainerLight = lightCyan
    static let primaryContainerDark = blueSlate

    // Secondary colors (Blue Slate for depth and creativity)
    static let secondaryLight = blueSlate
    static let secondaryDark = coolSteel

    // Secondary container colors
    static let secondaryContainerLight = lightBlue
    static let secondaryContainerDark = blueSlate

    // Background colors
    static let backgroundLight = lightCyan
    static let backgroundDark = jetBlack

    // Surface colors
    static let surfaceLight = Color(hex: "F5FEFF")    // Slightly brighter than background
    static let surfaceDark = Color(hex: "1A2428")     // Slightly lighter than background

    // Surface variant colors
    static let surfaceVariantLight = lightBlue
    static let surfaceVariantDark = blueSlate

    // MARK: - Adaptive Colors

    static var adaptivePrimary: Color {
        Color(UIColor { traitCollection in
            traitCollection.userInterfaceStyle == .dark ? UIColor(primaryDark) : UIColor(primaryLight)
        })
    }

    static var adaptiveBackground: Color {
        Color(UIColor { traitCollection in
            traitCollection.userInterfaceStyle == .dark ? UIColor(backgroundDark) : UIColor(backgroundLight)
        })
    }

    static var adaptiveSurface: Color {
        Color(UIColor { traitCollection in
            traitCollection.userInterfaceStyle == .dark ? UIColor(surfaceDark) : UIColor(surfaceLight)
        })
    }

    static var adaptiveSurfaceVariant: Color {
        Color(UIColor { traitCollection in
            traitCollection.userInterfaceStyle == .dark ? UIColor(surfaceVariantDark) : UIColor(surfaceVariantLight)
        })
    }

    static var adaptiveSecondary: Color {
        Color(UIColor { traitCollection in
            traitCollection.userInterfaceStyle == .dark ? UIColor(secondaryDark) : UIColor(secondaryLight)
        })
    }

    // MARK: - Adaptive Extended Colors

    static var adaptiveSuccess: Color {
        Color(UIColor { traitCollection in
            traitCollection.userInterfaceStyle == .dark ? UIColor(sageGreen) : UIColor(seaGlass)
        })
    }

    static var adaptiveWarning: Color {
        Color(UIColor { traitCollection in
            traitCollection.userInterfaceStyle == .dark ? UIColor(weatheredBrass) : UIColor(driftwood)
        })
    }

    static var adaptiveDisabled: Color {
        Color(UIColor { traitCollection in
            traitCollection.userInterfaceStyle == .dark ? UIColor(charcoal) : UIColor(mist)
        })
    }

    static var adaptiveActiveRoute: Color {
        coastalPath
    }

    static var adaptiveHistoricalRoute: Color {
        harborBlue
    }

    // MARK: - Semantic Helpers

    static var success: Color { seaGlass }
    static var successEmphasis: Color { sageGreen }
    static var warning: Color { driftwood }
    static var warningEmphasis: Color { weatheredBrass }
    static var disabled: Color { mist }
    static var disabledDark: Color { charcoal }

    // MARK: - Category Helpers

    static var neutralCategory: Color { stoneGray }
    static var waterCategory: Color { seafoamTint }
    static var eveningCategory: Color { duskPurple }
    static var morningCategory: Color { sunrisePeach }

    // MARK: - Route Helpers

    static var activeRoute: Color { coastalPath }
    static var historicalRoute: Color { harborBlue }
    static var alternativeRoute: Color { mistyLavender }

    // MARK: - Hex Initializer

    init(hex: String) {
        let hex = hex.trimmingCharacters(in: CharacterSet.alphanumerics.inverted)
        var int: UInt64 = 0
        Scanner(string: hex).scanHexInt64(&int)
        let a, r, g, b: UInt64
        switch hex.count {
        case 3: // RGB (12-bit)
            (a, r, g, b) = (255, (int >> 8) * 17, (int >> 4 & 0xF) * 17, (int & 0xF) * 17)
        case 6: // RGB (24-bit)
            (a, r, g, b) = (255, int >> 16, int >> 8 & 0xFF, int & 0xFF)
        case 8: // ARGB (32-bit)
            (a, r, g, b) = (int >> 24, int >> 16 & 0xFF, int >> 8 & 0xFF, int & 0xFF)
        default:
            (a, r, g, b) = (255, 0, 0, 0)
        }
        self.init(
            .sRGB,
            red: Double(r) / 255,
            green: Double(g) / 255,
            blue: Double(b) / 255,
            opacity: Double(a) / 255
        )
    }
}

// MARK: - UIColor Extension for Adaptive Colors

extension UIColor {
    convenience init(_ color: Color) {
        if #available(iOS 14.0, *) {
            self.init(color)
        } else {
            self.init(red: 0, green: 0, blue: 0, alpha: 1)
        }
    }
}

File: composeApp/src/main/kotlin/com/po4yka/trailglass/ui/theme/Color.kt
package com.po4yka.trailglass.ui.theme

import androidx.compose.ui.graphics.Color

// Silent Waters color palette
// Dewy blues and weathered gray merge into a hushed palette,
// like drifting fog on silent waters.

// MARK: - Base Palette Colors
val LightCyan = Color(0xFFE0FBFC)      // Delicate, sunlit blue tint
val LightBlue = Color(0xFFC2DFE3)       // Soft pastel blue
val CoolSteel = Color(0xFF9DB4C0)       // Cool, steely blue with hint of mist
val BlueSlate = Color(0xFF5C6B73)       // Blue with hints of cool authority
val JetBlack = Color(0xFF253237)        // Intense darkness with enigmatic appeal

// MARK: - Success/Confirmation Colors
val SeaGlass = Color(0xFF8BB5A1)        // Muted teal-green for success states
val SageGreen = Color(0xFF6A8E7F)       // Deeper confirmation color

// MARK: - Warning/Attention Colors
val Driftwood = Color(0xFFC9B896)       // Warm sand tone for warnings
val WeatheredBrass = Color(0xFFA89968)  // Deeper attention color

// MARK: - Map/Route Colors
val CoastalPath = Color(0xFF7A9CAF)     // Active route/current trip
val HarborBlue = Color(0xFF5C8AA8)      // Historical routes
val MistyLavender = Color(0xFFA8B5C7)   // Alternative path suggestion

// MARK: - Category/Tag Colors
val StoneGray = Color(0xFF8C979E)       // Neutral category
val SeafoamTint = Color(0xFFA8D5D8)     // Water-related trips
val DuskPurple = Color(0xFF8A90A6)      // Evening/night activities
val SunrisePeach = Color(0xFFD4B5A8)    // Morning activities

// MARK: - Disabled/Inactive States
val Mist = Color(0xFFD8E2E6)            // Disabled text/icons (light mode)
val Charcoal = Color(0xFF3A4449)        // Disabled text/icons (dark mode)

// MARK: - Data Visualization Gradient
val GradientStart = LightCyan           // (E0FBFC)
val GradientStep1 = LightBlue           // (C2DFE3)
val GradientStep2 = CoolSteel           // (9DB4C0)
val GradientStep3 = CoastalPath         // (7A9CAF)
val GradientEnd = BlueSlate             // (5C6B73)

// Primary colors (Cool Steel for tranquility and trust)
val PrimaryLight = CoolSteel
val OnPrimaryLight = Color(0xFFFFFFFF)
val PrimaryContainerLight = LightCyan
val OnPrimaryContainerLight = BlueSlate

val PrimaryDark = LightBlue
val OnPrimaryDark = JetBlack
val PrimaryContainerDark = BlueSlate
val OnPrimaryContainerDark = LightCyan

// Secondary colors (Blue Slate for depth and creativity)
val SecondaryLight = BlueSlate
val OnSecondaryLight = Color(0xFFFFFFFF)
val SecondaryContainerLight = LightBlue
val OnSecondaryContainerLight = JetBlack

val SecondaryDark = CoolSteel
val OnSecondaryDark = JetBlack
val SecondaryContainerDark = BlueSlate
val OnSecondaryContainerDark = LightCyan

// Tertiary colors (Accent variations for special elements)
val TertiaryLight = Color(0xFF7A8A92)   // Slightly warmer steel
val OnTertiaryLight = Color(0xFFFFFFFF)
val TertiaryContainerLight = Color(0xFFD4E4E8)
val OnTertiaryContainerLight = Color(0xFF2A3B42)

val TertiaryDark = Color(0xFFB8C8D0)
val OnTertiaryDark = JetBlack
val TertiaryContainerDark = Color(0xFF3F4E56)
val OnTertiaryContainerDark = Color(0xFFD4E4E8)

// Error colors (Muted red that harmonizes with the palette)
val ErrorLight = Color(0xFFBA1A1A)
val OnErrorLight = Color(0xFFFFFFFF)
val ErrorContainerLight = Color(0xFFFFDAD6)
val OnErrorContainerLight = Color(0xFF410002)

val ErrorDark = Color(0xFFFFB4AB)
val OnErrorDark = Color(0xFF690005)
val ErrorContainerDark = Color(0xFF93000A)
val OnErrorContainerDark = Color(0xFFFFDAD6)

// Background colors (Light cyan for serene light mode, jet black for dark)
val BackgroundLight = LightCyan
val OnBackgroundLight = JetBlack

val BackgroundDark = JetBlack
val OnBackgroundDark = LightCyan

// Surface colors
val SurfaceLight = Color(0xFFF5FEFF)    // Slightly brighter than background
val OnSurfaceLight = JetBlack
val SurfaceVariantLight = LightBlue
val OnSurfaceVariantLight = BlueSlate

val SurfaceDark = Color(0xFF1A2428)     // Slightly lighter than background
val OnSurfaceDark = LightCyan
val SurfaceVariantDark = BlueSlate
val OnSurfaceVariantDark = LightBlue

// Outline colors
val OutlineLight = CoolSteel
val OutlineDark = BlueSlate

// Surface tint
val SurfaceTintLight = PrimaryLight
val SurfaceTintDark = PrimaryDark
